src/prefect/new_flow_engine.py

```diff
@@ -355,11 +355,32 @@ def get_hooks(self, state: State, as_async: bool = False) -> Iterable[Callable]:
         if not flow_run:
             raise ValueError("Task run is not set")
 
+        enable_cancellation_and_crashed_hooks = (
```

Comment:
two related questions:
- why is this a setting?
- why is this set via env-var instead of using a setting object?

Reply:
This was introduced in https://github.com/PrefectHQ/prefect/pull/11026 when we started using the `Runner` to execute `on_cancellation` and `on_crashed` hooks. The runner is responsible this environment variable, and it isn't meant to be user-facing. It follows the pattern set by `PREFECT__FLOW_RUN_ID`, which is also set by the `Runner`.

Reply:
Ahhhh ok ok gotcha